test/modules/TestResultCoordinator/Reports/CountingReportMetadata.cs

```diff
@@ -2,13 +2,14 @@
 namespace TestResultCoordinator.Reports
 {
     using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
 
     class CountingReportMetadata : ITestReportMetadata
     {
         public CountingReportMetadata(string expectedSource, string actualSource, TestOperationResultType testOperationResultType, TestReportType testReportType)
         {
-            this.ExpectedSource = expectedSource;
-            this.ActualSource = actualSource;
+            this.ExpectedSource = Preconditions.CheckNonWhiteSpace(expectedSource, nameof(expectedSource));
```

Comment:
nit: I think if it is an internal class we don't need to check the preconditions here, it should be checked when they are read from the metadata

Reply:
leave it as-is; it is not harm to check.